Sprint 5/step 1

Ekaterina64 opened the pull request.

Ekaterina-ux committed:
Типизация стора

Ekaterina-ux committed:
Лента всех заказов и пользователя

gennady-bars commented on src/types/hooks.ts:
```diff
@@ -0,0 +1,9 @@
+	useSelector as selectorHook,
+} from 'react-redux'
+import { AppDispatch, RootState } from '../services/store'
+
+export const useAppDispatch = () => dispatchHook<AppDispatch>()
+export const useAppSelector: TypedUseSelectorHook<RootState> = selectorHook
```
* Отлично типизировали хуки `useSelector` и `useDispatch`
* Отлично типизировали стейт, диспатч

gennady-bars commented on src/components/app-header/app-header.tsx:
```diff
@@ -7,7 +7,7 @@ import {
 import { useSelector } from 'react-redux'
```
`import { useDispatch, useSelector } from 'react-redux';` такого больше не должно быть в проекте.  Хуки `useDispatch, useSelector` нужно типизировать **1 раз** в отдельном файле и заменить их во всем проекте на свои, а не из библиотеки `react-redux` .

gennady-bars commented on src/services/middleware/socket-middleware.ts:
```diff
@@ -0,0 +1,72 @@
+	onMessage: ActionCreatorWithPayload<any>
+}
+
+export const socketMiddleware = (wsActions: TwsActionTypes): Middleware => {
```
Хорошо сделали `socketMiddleware`

gennady-bars commented on src/services/middleware/socket-middleware.ts:
```diff
@@ -0,0 +1,72 @@
+}
+
+export const socketMiddleware = (wsActions: TwsActionTypes): Middleware => {
+	return store => {
```
В сокетах `store` типизировать можно вот так `(store: MiddlewareAPI<AppDispatch, RootState>)`

gennady-bars commented on src/pages/feed/feed.tsx:
```diff
@@ -0,0 +1,44 @@
+	useEffect(() => {
+		dispatch(feedWsConnect(`${NORMA_WSS}/all`))
+		return () => {
+			dispatch(feedWsDisconnect(1000))
```
Отлично, что закрываете соединение сокетов при покидании страниц

gennady-bars commented on src/components/burger-constructor/filling-list.tsx:
```diff
@@ -1,35 +1,27 @@
 	return (
 		<ul className={`${styles.list} custom-scroll mt-4 mb-4 pl-4 pr-1`}>
-			{fillings.map((fillingItem: IIngredient, index: number) => {
+			{fillings.map((fillingItem: TBurgerIngredient, index: number) => {
```
Вам больше **не нужно типизировать** аргументы в `map`, `filter`, `reduce` и тд, если массив уже 1 раз типизирован. Тайпскрипт возьмет тип автоматически. Код станет чище

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:



* Отлично типизировали хуки `useSelector` и `useDispatch`
* Отлично типизировали стейт, диспатч
* Отлично, что почти нет типов `any`
* Хорошо сделали `socketMiddleware`
* Отлично, что закрываете соединение сокетов при покидании страниц 
* Отличная логика с попапами заказов
* Хорошо типизировали редакс


но есть некоторые недочеты:




* Скрин https://disk.yandex.ru/i/ZLeQiBRPh0l9ig цифра из заказов под попапом отображается поверх попапа 
* `import { useDispatch, useSelector } from 'react-redux';` такого больше не должно быть в проекте.  Хуки `useDispatch, useSelector` нужно типизировать **1 раз** в отдельном файле и заменить их во всем проекте на свои, а не из библиотеки `react-redux` .  
* Вам больше **не нужно типизировать** аргументы в `map`, `filter`, `reduce` и тд, если массив уже 1 раз типизирован. Тайпскрипт возьмет тип автоматически. Код станет чище



**Можно лучше**

*  все `thunk`-функции нужно типизировать типом для `thunk`. Вам должны были показать, как его сделать. И еще вот ссылка на документацию по его типизации. https://redux.js.org/usage/usage-with-typescript#type-checking-redux-thunks

Ekaterina-ux committed:
Исправлено:
- Удалены useSelector и useDispatch
- Цифра доп заказов под модальным окном
- Убрана типизация аргументов из методов массива

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
